packages/api-rx/src/index.ts

```diff
@@ -0,0 +1,297 @@
+   *
+   * new Api().isReady.subscribe((api) => {
+   *   api.rpc.newHead().subscribe((header) => {
+   *     console.log(`new block #${header.blockNumber.toNumber()}`);
+   *   });
+   * });
+   * ```
+   */
+  constructor (wsProvider?: WsProvider) {
+    super();
+
+    this._rpc = new RpcRx(wsProvider);
+
+    this.isReady = from(this.init());
+  }
+
+  /**
+   * @description Observable that carries the connected state for the provider. Results in a boolean flag that is true/false based on the connectivity.
+   */
+  get isConnected (): Observable<boolean> {
+    return this.rpc.isConnected();
+  }
+
+  /**
+   * @description Contains the genesis Hash of the attached chain. Apart from being useful to determine the actual chain, it can also be used to sign immortal transactions.
+   */
+  get genesisHash (): Hash {
+    assert(!isUndefined(this._genesisHash), INIT_ERROR);
+
+    return this._genesisHash as Hash;
+  }
+
+  /**
+   * @description Contains all the raw rpc sections and their subsequent methods in the API as defined by the jsonrpc interface definitions.
+   * @example
+   * <BR>
+   *
+   * ```javascript
+   * api.rpc.chain
+   *   .newHead()
+   *   .subscribe((header) => {
+   *     console.log('new header', header);
+   *   });
+   * ```
+   */
+  get rpc (): RpcRx {
+    return this._rpc;
+  }
+
+  /**
+   * @description Yields the current attached runtime metadata. Generally this is only used to construct extrinsics & storage, but is useful for current runtime inspection.
+   */
+  get runtimeMetadata (): RuntimeMetadata {
+    assert(!isUndefined(this._runtimeMetadata), INIT_ERROR);
+
+    return this._runtimeMetadata as RuntimeMetadata;
+  }
+
+  /**
+   * @description Contains the version information for the current runtime.
+   */
+  get runtimeVersion (): RuntimeVersion {
+    assert(!isUndefined(this._runtimeVersion), INIT_ERROR);
+
+    return this._runtimeVersion as RuntimeVersion;
+  }
+
+  /**
+   * @description Contains all the chain state modules and their subsequent methods in the API. These are attached dynamically from the runtime metadata.
+   * @example
+   * <BR>
+   *
+   * ```javascript
+   * api.st.balances
+   *   .freeBalance(<accountId>)
+   *   .subscribe((balance) => {
+   *     console.log('new balance', balance);
+   *   });
+   * ```
+   */
+  get st (): QueryableStorage {
+    assert(!isUndefined(this._storage), INIT_ERROR);
+
+    return this._storage as QueryableStorage;
+  }
+
+  /**
+   * @description Contains all the extrinsic modules and their subsequent methods in the API. It allows for the construction of transactions and the submission thereof. These are attached dynamically from the runtime metadata.
+   * @example
+   * <BR>
+   *
+   * ```javascript
+   * api.tx.balances
+   *   .transfer(<recipientId>, <balance>)
+   *   .sign(<keyPair>, <accountNonce>, <blockHash (optional)>)
+   *   .send()
+   *   .subscribe((status) => {
+   *     console.log('tx status', status);
+   *   });
```

Comment:
could shorten to `.subscribe((status) => console.log('tx status', status));`?

Reply:
As above.